crates/dpapi-fuzzing/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+pub mod generator;
+
+use dpapi_core::{DecodeOwned, DecodeWithContextOwned, Encode, ReadCursor, WriteBuf};
+use dpapi_pdu::gkdi::{EcdhKey, FfcdhKey, FfcdhParameters, GetKey, GroupKeyEnvelope, KdfParameters, KeyIdentifier};
+use dpapi_pdu::rpc::{
+    AlterContext, AlterContextResponse, Bind, BindAck, BindNak, Command, ContextElement, ContextResult, EptMap,
+    EptMapResult, Fault, Floor, PacketFlags, Pdu, PduHeader, Request, Response, SecurityTrailer, SyntaxId,
+    VerificationTrailer,
+};
+
+use crate::generator::AnyStruct;
+
+pub fn round_trip(any: AnyStruct) {
+    let mut buf = WriteBuf::new();
+
+    if let Ok(name) = any.encode(&mut buf) {
+        let round_tripped_struct =
```

Comment:
note: Those are called "oracles". I think it would be slightly better to put all the oracles into a `oracle` module.